test/parallel/test-diagnostics-channel-tracing-channel-promise-early-exit.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const common = require('../common');
+const dc = require('diagnostics_channel');
+
+const channel = dc.tracingChannel('test');
+
+const handlers = {
+  start: common.mustNotCall(),
+  end: common.mustNotCall(),
+  asyncStart: common.mustNotCall(),
+  asyncEnd: common.mustNotCall(),
+  error: common.mustNotCall()
+};
+
+// While subscribe occurs _before_ the promise resolves,
+// no async events should be published.
+channel.tracePromise(() => {
```

Comment:
```suggestion
channel.tracePromise(common.mustCall(() => {
```
You could even add asserts that this/arguments/return values are correctly forwarded.
Similar of the other APIs. But feel free to ignore and keep as is.